fluent-bundle/src/types.ts

```diff
@@ -104,37 +117,134 @@ export class FluentNumber extends FluentType<number> {
+> {
   /** Options passed to `Intl.DateTimeFormat`. */
   public opts: Intl.DateTimeFormatOptions;
 
+  static supportsValue(value: any): value is ConstructorParameters<typeof Temporal.Instant>[0] {
+    if (typeof value === "number") return true;
+    if (value instanceof Date) return true;
+    if (value instanceof FluentType) return FluentDateTime.supportsValue(value.valueOf());
+    // Temporary workaround to support environments without Temporal
+    if ('Temporal' in globalThis) {
+      if (
+        // @ts-ignore
+        value instanceof Temporal.Instant        || // @ts-ignore
+        value instanceof Temporal.PlainDateTime  || // @ts-ignore
+        value instanceof Temporal.PlainDate      || // @ts-ignore
+        value instanceof Temporal.PlainMonthDay  || // @ts-ignore
+        value instanceof Temporal.PlainTime      || // @ts-ignore
+        value instanceof Temporal.PlainYearMonth || // @ts-ignore
+        value instanceof Temporal.ZonedDateTime
+      ) {
+        return true;
+      }
+    }
+    return false
+  }
+
   /**
    * Create an instance of `FluentDateTime` with options to the
    * `Intl.DateTimeFormat` constructor.
    *
    * @param value The number value of this `FluentDateTime`, in milliseconds.
    * @param opts Options which will be passed to `Intl.DateTimeFormat`.
    */
-  constructor(value: number, opts: Intl.DateTimeFormatOptions = {}) {
+  constructor(
+    value:
+      | number
+      | Date
+      | Temporal.Instant
+      | Temporal.PlainDateTime
+      | Temporal.PlainDate
+      | Temporal.PlainMonthDay
+      | Temporal.PlainTime
+      | Temporal.PlainYearMonth
+      | Temporal.ZonedDateTime
+      | FluentDateTime
+      | FluentType<number>,
+    opts: Intl.DateTimeFormatOptions = {}
+  ) {
+    // unwrap any FluentType value, but only retain the opts from FluentDateTime
+    if (value instanceof FluentDateTime) {
+      opts = { ...value.opts, ...opts };
+      value = value.value;
+    } else if (value instanceof FluentType) {
+      value = value.valueOf();
+    }
+
+    if (typeof value === "object") {
+      // Intl.DateTimeFormat defaults to gregorian calendar, but Temporal defaults to iso8601
+      if ('calendarId' in value) {
+        if (opts.calendar === undefined) {
+          opts = { ...opts, calendar: value.calendarId };
+        } else if (opts.calendar !== value.calendarId && 'withCalendar' in value) {
+          value = value.withCalendar(opts.calendar);
+        }
+      }
+
+      // Temporal.ZonedDateTime is timezone aware
+      if ('timeZoneId' in value) {
+        if (opts.timeZone === undefined) {
+          opts = { ...opts, timeZone: value.timeZoneId };
+        } else if (opts.timeZone !== value.timeZoneId && 'withTimeZone' in value) {
+          value = value.withTimeZone(opts.timeZone);
+        }
+      }
+
+      // Temporal.ZonedDateTime cannot be formatted directly
+      if ('toInstant' in value) {
+        value = value.toInstant();
+      }
```

Comment:
This automatic conversion is also a bad idea. As your comment notes, zoned datetimes are not formattable at least for now, so let's stick to that in Fluent as well.